vnext/local-cli/config/dependencyConfig.js

```diff
@@ -1,102 +1,244 @@
+      csNamespaces: [],        // (opt) Array of cs namespaces, ie: 'MyModule', to be transformed into 'using MyModule;'
+      csPackageProviders: [],  // (opt) Array of fully qualified cs IReactPackageProviders, ie: 'MyModule.ReactPackageProvider'
+    },
+  ],
+  nugetPackages: [ // (opt) Array of nuget packages including native modules that must be added as a dependency to the consuming app. It can be empty, but by its nature it can't be calculated
+    {
+      packageName: string,     // (req) Name of the nuget package to install
+      packageVersion: string,  // (req) Version of the nuget package to install
+      cppHeaders: [],          // (req) Array of cpp header include lines, ie: 'winrt/NugetModule.h', to be transformed into '#include <winrt/NugetModule.h>'
+      cppPackageProviders: [], // (req) Array of fully qualified cpp IReactPackageProviders, ie: 'NugetModule::ReactPackageProvider'
+      csNamespaces: [],        // (req) Array of cs namespaces, ie: 'NugetModule', to be transformed into 'using NugetModule;'
+      csPackageProviders: [],  // (req) Array of fully qualified cs IReactPackageProviders, ie: 'NugetModule.ReactPackageProvider'
+    },
+  ],
+}
 
-  if (!sourceDir) {
+Example react-native.config.js for a 'MyModule':
+
+module.exports = {
+  dependency: {
+    platforms: {
+      windows: {
+        sourceDir: 'windows',
+        solutionFile: 'MyModule.sln',
+        projects: [
+          {
+            projectFile: 'MyModule\\MyModule.vcxproj',
+            directDependency: true,
+          }
+        ],
+      },
+    },
+  },
+};
+
+*/
+
+/**
+ * Gets the config of any RNW native modules under the target folder.
+ * @param {string} folder The absolute path to the target folder.
+ * @param {object} userConfig A manually specified override config.
+ * @return {object} The config if any RNW native modules exist.
+ */
+function dependencyConfigWindows(folder, userConfig = {}) {
+  if (userConfig === null) {
     return null;
   }
 
-  var packageName = null;
-  const packageIDL = findPackageProviderIDL(sourceDir);
-  if (packageIDL){
-    packageName = parsePackageIDLFile(packageIDL);
+  const usingManualProjectsOverride =
+    'projects' in userConfig && Array.isArray(userConfig.projects);
+
+  const usingManualNugetPackagesOverride =
+    'nugetPackages' in userConfig && Array.isArray(userConfig.nugetPackages);
+
+  var result = {
+    folder,
+    projects: usingManualProjectsOverride ? userConfig.projects : [],
+    nugetPackages: usingManualNugetPackagesOverride
+      ? userConfig.nugetPackages
+      : [],
+  };
+
+  var sourceDir = null;
+  if (usingManualProjectsOverride && result.projects.length > 0) {
+    // Manaully provided projects, so extract the sourceDir
+    if (!('sourceDir' in userConfig)) {
+      sourceDir =
+        'Error: Source dir is required if projects are specified, but it is not specified in react-native.config.';
```

Comment:
throw this string instead of returning it? (here and below)

Reply:
I started with that, but unfortunately, the structure of the config command is that it disables logging and returns a JSON blob output or an error. Having a single error thrown breaks the whole command, which means you also break the entire CLI and it's commands, even for other platforms. That's a crappy side effect, and it doesn't even catch when there are multiple errors (say if you have errors in your app config and there's errors in some dependency's config).

The only other option is to fail silently by leaving filed null, then having the caller of `react-native config` go figure out why it was wrong, or set the field to an error that we can see and process later in our consumption of the config.